src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Serialization/CustomConverterTests/CustomConverterTests.Callback.cs

```diff
@@ -133,7 +133,7 @@ public static void ConverterWithReentryFail()
 
             ex = Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<TopLevelPocoWithNoConverter>(Json, options));
             Assert.Contains(typeof(int[,]).ToString(), ex.ToString());
-            Assert.Contains(typeof(ChildPocoWithNoConverterAndInvalidProperty).ToString(), ex.ToString());
```

Comment:
This assertion is in fact incorrect, since the exception emanates from the converter factory of the parent type. The updated logic now surfaces the correct type name.